c7n/schema.py

```diff
@@ -69,12 +72,40 @@ def validate(data, schema=None):
         logging.exception(
             "specific_error failed, traceback, followed by fallback")
 
+    extra_errs = validate_policy_dt_parse(data)
+    if extra_errs:
+        return extra_errs
+
     return list(filter(None, [
         errors[0],
         best_match(validator.iter_errors(data)),
     ]))
 
 
+def validate_policy_dt_parse(data):
+
```

Comment:
per previous comment, this method really belongs inside of the policy class, its feels like an encapsulation violation as is.